File: practice23/11.cpp


//https://www.nowcoder.com/practice/f7500a858bde41108f01286f57f554f3?tpId=98&tqId=32878&tPage=3&rp=3&ru=/ta/2019test&qru=/ta/2019test/question-ranking
#include<iostream>
#include<string>
using namespace std;

int main()
{
	string str;
	int n;
	cin >> str >> n;
	int m = str.size();
	int x = 0;
	if (n <= 0 || n > m)
	{
		cout << -1 << endl;
	}
	else
	{
		while (1)
		{
			for (int i = x; i < x + n; i++)
			{
				cout << str[i];
			}
			x++;
			if (x + n > m)
			{
				cout << endl;
				break;
			}
			else
			{
				cout << " ";
			}
		}
	}
	system("pause");
	return 0;
}

File: practice28/11.java


//https://www.nowcoder.com/practice/a4771283f1c9435d9aeb3045d55dc030?tpId=98&tqId=33049&tPage=12&rp=12&ru=/ta/2019test&qru=/ta/2019test/question-ranking
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner=new Scanner(System.in);
        int n=scanner.nextInt();
        int []pt=new int[n];
        int sum=0;
        for(int i=0;i<n;i++){
            pt[i]=scanner.nextInt();
            sum=sum+pt[i];
        }
        if(sum%n==1){
            System.out.println(-1);
            System.exit(0);
        }
        int diff=0;
        int aver=sum/n;
        int count=0;
        for(int i=0;i<n;i++){
            diff=pt[i]-aver;
            if(Math.abs(diff)%2==1){
                System.out.println(-1);
                System.exit(0);
            }
            if(diff>0){
                count+=diff/2;
            }
        }
        System.out.println(count);
    }
}
